src/actions/flashMessageActions.js

```diff
@@ -1,23 +1,64 @@
 'use strict';
 
-import * as types from './actionTypes';
+import Noty from 'noty';
 
-export function flashRemove(flashMessage) {
-  return { type: types.FLASH_REMOVE, flashMessage };
-}
+// messageType affects the visual markup to distinguish severity
+export const messageType = {
+  ERROR: 'error',
+  INFO: 'info',
+  SUCCESS: 'success',
+  WARNING: 'warning',
+};
+
+// Display duration for a message.
+// bool false means forever.
+// integers are a duration in milliseconds.
+export const messageTTL = {
+  FOREVER: false,
+  SHORT: 1500,
+  MEDIUM: 3000,
+  LONG: 10000,
+  VERYLONG: 30000,
+};
 
-export function flashAdd(flashMessage) {
-  return function(dispatch) {
-    dispatch({ type: types.FLASH_ADD, flashMessage });
+export class FlashMessage {
+  constructor(text, type, ttl, subtext) {
```

Comment:
Why create a new class and not reuse the function signature but extend params?

Something like
```javascript
export function flashAdd(text, type, ttl, subtext) {

    this.noty = new Noty({
      type: type,
      text: this.text,
      timeout: this.timeout,
      theme: 'bootstrap-v3',
      layout: 'topRight',
      visibilityControl: true,
      animation: {
        close: 'flash_message_close',
      },
    });

    this.noty.show();
}
```

Reply:
Mainly because I find it very expressive to say "new Something" to "create" a something. Beyond that, the difference is small.